### PR TITLE
Prevent a crash by exiting here

### PR DESCRIPTION
**Pull request by JohnAFernandez**
And make an assertion more explicit, just in case.

Fixes a crash for observers with certain ship table setups.

### CODE REVIEW

File: code/hud/hudreticle.cpp [hunk 1]
```diff
@@ -309,12 +309,19 @@ ship_info *sip = &Ship_info[Player_ship->ship_info_index];
 }
 
 void HudGaugeReticle::getFirepointStatus() {
+
+	// allow the firepoint status to be empty when a multiplayer observer
+	// this is not a bug, the observer will simply *not* have any firepoints.
+	if (Objects[Player->objnum].type == OBJ_OBSERVER) {
+		return;
```
Comment (jg18) [inline code/hud/hudreticle.cpp:316]: Should we also assert here that `Game_mode & GM_MULTIPLAYER`?

File: code/hud/hudreticle.cpp [hunk 2]
```diff
@@ -309,12 +309,19 @@ ship_info *sip = &Ship_info[Player_ship->ship_info_index];
...
+		return; 
+	}
+
 	//First, get the player ship
 	ship_info* sip;
 	ship* shipp;
 	polymodel* pm;
 
-	Assert(Objects[Player->objnum].type == OBJ_SHIP);
+	Assertion(Objects[Player->objnum].type == OBJ_SHIP, "HudGaugeReticle::getFirepointStatus was passed an invalid object type of %d.", Objects[Player->objnum].type);
```
Comment (jg18) [inline code/hud/hudreticle.cpp:324]: How about adding a "Please report"?

**Review by jg18 — COMMENTED** [review]
Adding a couple thoughts.

File: code/hud/hudreticle.cpp (hunk 1 above)
Comment (JohnAFernandez) [reply code/hud/hudreticle.cpp:316]: That is a great idea! :D

File: code/hud/hudreticle.cpp (hunk 2 above)
Comment (JohnAFernandez) [reply code/hud/hudreticle.cpp:324]: Yes, of course!

**Review by EatThePath — APPROVED** [review]
Looks all good to me

**Comment by JohnAFernandez** [comment]
Thanks @EatThePath

**Pull request merged**